corehq/apps/geospatial/forms.py

```diff
@@ -164,4 +208,14 @@ def clean(self):
             if not cleaned_data['target_group_count']:
                 raise ValidationError(_("Value for target group count required"))
 
+        algorithm = cleaned_data.get('selected_disbursement_algorithm')
```

Comment:
nit: should we also add a check here to ensure the domain has the feature flag enabled?
I understand its not that critical so even if one bypasses the check on the form and save the configuration with this new algorithm it is not critical but good to do add such checks as a practice.

Reply:
>  the form validation fails if the flag is not enabled, because the road_network_algorithm is not added as a valid option for the selected_disbursement_algorithm input.

This is good enough for me. Thanks.